analyzers/src/SonarAnalyzer.CSharp/Rules/PropertyNamesShouldNotMatchGetMethods.cs

```diff
@@ -27,75 +27,58 @@
+        private const string DiagnosticId = "S4059";
+        private const string MessageFormat = "Change either the name of property '{0}' or the name of method '{1}' to make them distinguishable.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
                 {
-                    if (!(c.SemanticModel.GetDeclaredSymbol(c.Node) is INamedTypeSymbol classSymbol))
+                    if (c.SemanticModel.GetDeclaredSymbol(c.Node) is not INamedTypeSymbol typeSymbol)
                     {
                         return;
                     }
+                    var typeMembers = typeSymbol.GetMembers().Where(SymbolHelper.IsPubliclyAccessible);
+                    var properties = typeMembers.OfType<IPropertySymbol>().Where(property => !property.IsOverride).ToArray();
+                    var methods = typeMembers.OfType<IMethodSymbol>().ToArray();
 
-                    var classMembers = classSymbol.GetMembers().Where(SymbolHelper.IsPubliclyAccessible);
-                    var properties = classMembers.OfType<IPropertySymbol>().Where(property => !property.IsOverride);
-                    var methods = classMembers.OfType<IMethodSymbol>().ToList();
-
-                    foreach (var collidingMembers in GetCollidingMembers(properties, methods))
+                    foreach (var collidingMembers in CollidingMembers(properties, methods))
                     {
                         var propertyIdentifier = collidingMembers.Item1;
                         var methodIdentifier = collidingMembers.Item2;
-
-                        c.ReportIssue(Diagnostic.Create(
-                            rule,
-                            propertyIdentifier.GetLocation(),
-                            additionalLocations: new[] { methodIdentifier.GetLocation() },
-                            messageArgs: new[] { propertyIdentifier.ValueText, methodIdentifier.ValueText }));
+                        c.ReportIssue(Diagnostic.Create(Rule, propertyIdentifier.GetLocation(), new[] { methodIdentifier.GetLocation() }, propertyIdentifier.ValueText, methodIdentifier.ValueText));
```

Comment:
For this single rule or for each rule using location?

Reply:
> I think this is a general problem in many analyzers

I think that this should be discussed with source generators and older source generating tools (like dbml) in mind. We may face this problem in more places in the future because the compiler team is about to allow `partial` on more occasions (e.g. properties) in the future because SGs would benefit greatly from such a move. So a unfied/central solution would be the best thing IMO.

Reply:
You can create a reproducer that demonstrates the problem on any rule. Create an issue for it and suggest the proper implementation there (that would probably end up as some kind of extension method)

Reply:
I added #5618.